src/engine/TransitivePath.cpp

```diff
@@ -68,21 +69,41 @@ std::string TransitivePath::asString(size_t indent) const {
 // _____________________________________________________________________________
 std::string TransitivePath::getDescriptor() const {
   std::ostringstream os;
-  if (_leftSideTree != nullptr) {
-    os << "TransitivePath left is subtree, rightCol " << _rightSubCol;
-  } else if (_rightSideTree != nullptr) {
-    os << "TransitivePath leftCol " << _leftSubCol << " right is subtree";
+  os << "TransitivePath ";
+  // If not full transitive hull, show interval as [min, max].
+  if (_minDist > 1 || _maxDist < std::numeric_limits<size_t>::max()) {
+    os << "[" << _minDist << ", " << _maxDist << "] ";
+  }
+  // Left variable or entity name.
+  if (_leftIsVar) {
+    os << _leftColName;
   } else {
-    os << "TransitivePath leftCol " << _leftSubCol << " rightCol "
-       << _rightSubCol;
+    auto optional = getIndex().idToOptionalString(_leftValue);
+    if (optional) {
+      os << optional.value();
+    } else {
+      os << "#" << _leftValue;
+    }
```

Comment:
`os << getIndex().idToOptionalString(_leftValue).value_or("#" + std::to_string(_leftValue))`
https://en.cppreference.com/w/cpp/utility/optional/value_or

(If you want to, you can make two lines from this, I am sorry, I did not directly tell you this possibility:)

Reply:
Perfect, thanks!